cg/cli/set.py

```diff
@@ -186,9 +188,58 @@ def microbial_order(context, apptag, order_id, user_signature):
         click.echo(click.style(f"order not found: {order_id}", fg='yellow'))
         context.abort()
 
-    for sample_obj in microbial_order_obj.microbial_samples:
-        context.invoke(microbial_sample, sample_id=sample_obj.internal_id,
-                       user_signature=user_signature, apptag=apptag)
+    if apptag:
+        for sample_obj in microbial_order_obj.microbial_samples:
+            context.invoke(microbial_sample, sample_id=sample_obj.internal_id,
+                           user_signature=user_signature, apptag=apptag)
+
+    if ticket:
+        if microbial_order_obj.ticket_number == ticket:
+            click.echo(click.style(f"Order {microbial_order_obj.internal_id} already has the "
+                                   f"ticket {ticket}", fg='yellow'))
+            return
+
+        comment = f"Ticket changed from" \
+            f" {microbial_order_obj.ticket_number} to " \
+            f"{ticket} by {user_signature}"
+        microbial_order_obj.ticket_number = ticket
+        click.echo(click.style(f"Ticket for {microbial_order_obj.internal_id} set to "
```

Comment:
Technically, this is not set until it is commit. Is there any way we can gather all messages and display them after commit?

Reply:
implemented